src/GameLogic/DataStructures/ActionSequence.ts

```diff
@@ -0,0 +1,24 @@
+import {PlayerAction} from "../Enums";
+
+export class ActionSequence {
+    private _actionSequence: PlayerAction[];
+    private _step: number = 0;
+
+	constructor(actionSequence: PlayerAction[] = [])
+	{
+		this._actionSequence = actionSequence;
+    }
+    
+    public push(action: PlayerAction): void {
+		this._actionSequence.push(action);
+    }
+    
+    public getNext(): PlayerAction {
```

Comment:
Return type is missing `|undefined`, since you return undefined in case there is no action remaining

Reply:
Trying to run the tests flagged this as an issue.